js/main.js

```diff
@@ -0,0 +1,102 @@
+
+/* Function that test if our API works */
+
+/* code for <ul> element that holds our list items */
+var $productListing = document.querySelector('#product-listing');
+
+var xhr = new XMLHttpRequest();
+xhr.open('GET', 'http://makeup-api.herokuapp.com/api/v1/products.json');
+xhr.responseType = 'json';
+
+xhr.addEventListener('load', function () {
+
+  event.preventDefault();
+
+  /* log the xhr status & response to ensure it works
+      console.log(xhr.status);
+      console.log(xhr.response); */
+
+  for (var i = 0; i < xhr.response.length; i++) {
+
+    var newListing = {
+      name: xhr.response[i].name,
+      price: xhr.response[i].price,
+      entryId: xhr.response[i].id,
+      image: xhr.response[i].image_link
+
+    };
+    /* To account for broken links & values in API:
+       This condition states that if the price is not strictly equal to zero
+       or null, then to append it to our listing */
+    var makeUpProducts = renderListing(newListing);
```

Comment:
I would put this line into the below conditional as well, there's no point generating a DOM tree if you don't end up using it.

Reply:
That makes sense! I will go ahead and fix this